botfront/imports/ui/components/settings/GitSettings.jsx

```diff
@@ -41,22 +48,42 @@ class GitSettings extends React.Component {
                     this.setState({ saved: true });
                     this.successTimeout = setTimeout(() => {
                         this.setState({ saved: false });
+                        this.hideCreds(true)
                     }, 2 * 1000);
+                    
                 }
                 this.setState({ saving: false });
+                
             }),
         );
     };
 
+    gitSettingsEmpty = (gitSettings) => { 
+        if (!gitSettings) return true
+        return Object.values(gitSettings).every(val => !val)
+    }
+   
+
     renderGitSettings = () => {
         const { gitSettings, projectId } = this.props;
-        const { saving, saved } = this.state;
+        const { saving, saved, hidden } = this.state;
         const bridge = new SimpleSchema2Bridge(GitSettingsSchema);
         const hasWritePermission = can('git-credentials:w', projectId);
+        const obfuscation = { 
+        //we use this obfuscation because it matches the  validation regex, thus no error are shown when obfuscating
+        gitString: 'https://******:******@******.******#******', 
```

Comment:
since the errors field is hidden when in "hide" mode this field can just be **********************

Don't forget to update the test with the new value as well 🙂

Reply:
even when the error field is hidden, the field it self is tinted red 
![image](https://user-images.githubusercontent.com/27060431/108275707-bd258680-7144-11eb-925d-0ab6e6cca819.png)
So using https://******:******@******.******#****** is the easiest way prevent the red from showing on the field

Reply:
ah ok, I figured out the better way to do this, uniform fields have a type prop that, when set to password turns the characters into *******

so type={hidden ? 'password' : null} seems to fix this issue

and we wont need the obfuscation model anymore which might make it a bit cleaner

Reply:
Nice ! Thanks

Reply:
Actually it  works perfectly for the inputs but not for the textarea  used for the private key. There might be a way to get the same behavior but from the stackoverflows it's seems more hacky than the unclean obfuscation model